contents/testing/security-testing.md

```diff
@@ -0,0 +1,144 @@
+
+Due to the logical limitations of security testing however, passing security testing is not an indication that no flaws exist or that the system adequately satisfies the security requirements[^4]. Why have security testing then? In brief, a system would be more secure with security testing than without.
+
+<box id="info">:fas-info-circle: There is more to the concept of security than the working definition used here. If you would like to learn more about security in-depth, please see the [references](#references).</box>
+
+## Security Testing in Action
+
+The consensus is that it is much more costly to include security testing after software implementation or deployment[^5]. It is necessary then to involve security testing in all phases of a Software Development Life Cycle ([SDLC](https://se-education.org/se-book/processModels/)). A conceptual example of such an involvement is as follows (taken from [Guru99.com](https://www.guru99.com/what-is-security-testing.html#3)):
+
+<center>
+
+<pic src="security-testing/Sec-Test-Fig-1.png" alt="Example">
+Figure 1 - Example of Security Testing during a SDLC.
+</pic>
+</center>
+
+<br>
+
+To understand this in more detail, we can deconstruct the application of security testing into its manifestation in different stages of software development. The following series of steps taken from the OWASP Testing Guide[^1] provide general applications of security testing in a generic SDLC.
+
+### Step 1: Before Development
+
+#### 1.1 Define an SDLC
+
+Before application development starts an adequate SDLC must be defined where security is inherent at each stage. 
+
+#### 1.2 Review Policies and Standards
+
+Ensure that there are appropriate policies, standards, and documentation in place. Documentation is extremely important as it gives development teams guidelines and policies that they can follow. An example of one such documentation is the OWASP Secure Coding Practices Quick Refernce Guide[^6].
+
+People can only do the right thing if they know what the right thing is.
+
+If the application is to be developed in Java, it is essential that there is a Java secure coding standard. If the application is to use cryptography, it is essential that there is a cryptography standard. No policies or standards can cover every situation that the development team will face. By documenting the common and predictable issues, there will be fewer decisions that need to be made during the development process.
+
+### Step 2: During Definition and Design
+
+#### 2.1 Review Security Requirements
+
+Security requirements define how an application works from a security perspective. It is essential that the security requirements are tested. Testing in this case means testing the assumptions that are made in the requirements and testing to see if there are gaps in the requirements definitions.
```

Comment:
... are tested. In particular, assumptions that are made in the requirements are tested while checking if there are any gaps in the requirement definitions

Perhaps wording it this way would reduce the number of uses for the term "testing"? However, it is a suggestion, feel free to modify if you think it might be better